framework/src/base/BlockRestrictable.C

```diff
@@ -27,6 +27,8 @@ InputParameters validParams<BlockRestrictable>()
   // Add the user-facing 'block' input parameter
   params.addParam<std::vector<SubdomainName> >("block", "The list of block ids (SubdomainID) that this object will be applied");
 
+  params.addPrivateParam<bool>("delay_initialization", false);
```

Comment:
We could do better I suppose. Parameters aren't really documented either unless people write Doxygen around their signatures.

Continuing off-topic, I kind of like the idea of unifying our Interface headers as well. The reason is similar to that for unifying our object headers. If I were to change the constructor for this class I'd have to change it everywhere it's used. Right now, I can slip a new parameter in its validParams method and then start using it. Yes, applications are indeed making use of our Interfaces in some places so this problem is real. It's just not nearly as widespread.